### PR TITLE
Tests & ESLint Improvements

### LINKED ISSUE #6989: Tests & ESLint Improvements
- [x] Reduce unit testing time on local
- [x] Reduce ESLint time on local

and after that https://github.com/web3/web3.js/issues/6966 compare results on local with older commit.

### PR DESCRIPTION
**Pull request by avkos**
## Description

Please include a summary of the changes and be sure to follow our [Contribution Guidelines](https://github.com/web3/web3.js/blob/4.x/.github/CONTRIBUTING.md).

Fixes #6989 

### Results
LINT from 2m 10s to **44s** (lint only changes, cache). locally - few seconds
UNIT from 3m 42s to **2m 48s** (type check off)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] I have selected the correct base branch.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran `npm run lint` with success and extended the tests and types if necessary.
- [ ] I ran `npm run test:unit` with success.
- [ ] I ran `npm run test:coverage` and my test cases cover all the lines and branches of the added code.
- [ ] I ran `npm run build` and tested `dist/web3.min.js` in a browser.
- [ ] I have tested my code on the live network.
- [ ] I have checked the Deploy Preview and it looks correct.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.
- [ ] I have linked Issue(s) with this PR in "Linked Issues" menu.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -65,12 +65,65 @@ jobs:
...
                     node-version: 18
-            -   uses: actions/cache/restore@v3
+            -   uses: actions/cache/restore@v4
                 with:
                     path: ./
                     key: web3-18-${{github.sha}}
             -   run: npx ts-node scripts/init.ts
+            # restore cache for each package from main branch
+            -   name: Restore eslint caches
+                #  if: github.event_name == 'push' && github.ref == 'refs/heads/4.x'
```
Comment (luu-alex) [inline .github/workflows/build.yml]: should be uncommented?

File: packages/web3-core/package.json [hunk 2]
```diff
@@ -30,11 +30,11 @@
 		"build:esm": "tsc --build tsconfig.esm.json && echo '{\"type\": \"module\"}' > ./lib/esm/package.json",
 		"build:types": "tsc --build tsconfig.types.json",
 		"build:check": "node -e \"require('./lib')\"",
-		"lint": "eslint --ext .js,.ts .",
+		"lint": "eslint --cache --cache-strategy content --ext .ts .",
 		"lint:fix": "eslint --fix --ext .js,.ts .",
 		"format": "prettier --write '**/*'",
 		"test": "jest --config=./test/unit/jest.config.js",
-		"test:coverage:unit": "jest --config=./test/unit/jest.config.js --coverage=true --coverage-reporters=text",
+		"test:coverage:unit": "jest --config=./test/unit/jest.config.js  --coverage=true --coverage-reporters=text",
```
Comment (luu-alex) [inline packages/web3-core/package.json]:
```suggestion
		"test:coverage:unit": "jest --config=./test/unit/jest.config.js --coverage=true --coverage-reporters=text",
```

Comment (luu-alex) [reply packages/web3-core/package.json]: unneccesary extra space

**Review by luu-alex — APPROVED** [review]

**Comment by luu-alex** [comment]
lgtm other than the comments ^

File: .github/workflows/build.yml (hunk 1 above)
Comment (avkos) [reply .github/workflows/build.yml]: no. we need to restore everytime when we check. remove comment

File: packages/web3-core/package.json (hunk 2 above)
Comment (avkos) [reply packages/web3-core/package.json]: fixed in all packages

**Review by jdevcs — APPROVED** [review]

**Pull request merged**